crates/rspack_core/src/utils/task_loop.rs

```diff
@@ -95,7 +92,7 @@ pub fn run_task_loop_with_event<Ctx: 'static>(
         }
         TaskType::Sync => {
           // merge sync task result directly
-          match task.sync_run(ctx) {
+          match task.sync_run(ctx).await {
```

Comment:
the sync_run actually means run in single thread, so change it to async to avoid call block_on in it, will change the name to `run_in_main_thread` maybe in following pr